src/Moryx.AbstractionLayer.Products.Endpoints/ProductManagementController.cs

```diff
@@ -0,0 +1,332 @@
+            {
+                productModels.Add(_productConverter.ConvertProduct(t, false));
+            }
+            return productModels.ToArray();
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{id}")]
+        public ActionResult<ProductModel> GetType(long id)
+        {
+            if (id == 0)
+                return BadRequest($"Id was 0");
+            var productType = _productManagement.LoadType(id);
+            if (productType == null)
+                return NotFound();
+            return _productConverter.ConvertProduct(productType, false);
+        }
+
+        [HttpDelete]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{id}")]
+        public ActionResult<bool> DeleteType(long id)
+        {
+            var result = _productManagement.DeleteProduct(id);
+            if (!result)
+                return NotFound();
+            return result;
+        }
+
+        [HttpPut]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Route("producttypes/{id}")]
+        public ActionResult<long> UpdateType(long id, ProductModel modifiedType)
+        {
+            if (modifiedType == null)
+                return BadRequest($"Modified product type was null");
+            var type = _productManagement.LoadType(id);
+            if (type == null)
+                return BadRequest($"No product type with id {modifiedType.Id} was found");
+            type = _productConverter.ConvertProductBack(modifiedType, (ProductType) type);
+            return _productManagement.SaveType(type);
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{identifier}/{revision}")]
+        public ActionResult<ProductModel> GetType(string identifier, short revision)
+        {
+
+            if (identifier == null)
+                return BadRequest($"Identity was null");
+            var identity = new ProductIdentity(WebUtility.HtmlEncode(identifier), revision);
+            var productType = _productManagement.LoadType(identity);
+            if (productType == null)
+                return NotFound();           
+            return _productConverter.ConvertProduct(productType, false);
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("producttypes/{newIdentifier}/{newRevision}")]
```

Comment:
I would prefer to post to "types", since you are simply saving a new type. For duplication you could include an optional query argument with the template/source.

Reply:
I think our Facade can only update productTypes not save new ones. For saving a new one an importer is needed

Reply:
Importers are recommended for more complex type creation, but you can pass a new objekt to the Save method of the facade.

Anyway since duplication requires a current template object, how about `[HttpPost("types/{id}/duplicate")]`?